components/ipfs/ipfs_utils.cc

```diff
@@ -89,38 +87,36 @@ bool IsValidCID(const std::string& cid) {
   return base::StartsWith(cid, kCIDv0Prefix);
 }
 
-bool IsIpfsDisabledByPolicy(content::BrowserContext* context) {
-  DCHECK(context);
-  PrefService* prefs = user_prefs::UserPrefs::Get(context);
-  return prefs->FindPreference(kIPFSEnabled) &&
-         prefs->IsManagedPreference(kIPFSEnabled) &&
-         !prefs->GetBoolean(kIPFSEnabled);
-}
-
-bool IsIpfsEnabled(content::BrowserContext* context) {
-  DCHECK(context);
-  if (context->IsOffTheRecord() || IsIpfsDisabledByPolicy(context) ||
-      !base::FeatureList::IsEnabled(ipfs::features::kIpfsFeature)) {
-    return false;
-  }
-
-  return true;
-}
-
-bool IsIpfsResolveMethodDisabled(content::BrowserContext* context) {
-  DCHECK(context);
+bool IsIpfsResolveMethodDisabled(PrefService* prefs) {
+  DCHECK(prefs);
 
   // Ignore the actual pref value if IPFS feature is disabled.
-  if (!IsIpfsEnabled(context)) {
+  if (IsIpfsDisabled(prefs)) {
     return true;
   }
 
-  PrefService* prefs = user_prefs::UserPrefs::Get(context);
   return prefs->FindPreference(kIPFSResolveMethod) &&
          prefs->GetInteger(kIPFSResolveMethod) ==
              static_cast<int>(ipfs::IPFSResolveMethodTypes::IPFS_DISABLED);
 }
 
+bool IsIpfsMenuEnabled(PrefService* prefs) {
+  return !ipfs::IsIpfsDisabled(prefs) && ipfs::IsLocalGatewayConfigured(prefs);
+}
+
+bool IsIpfsDisabled(PrefService* prefs) {
```

Comment:
I would try to improve naming, because currently we have `IsIpfsDisabled` here and `IsIpfsEnabled` in the factory, and these two check different things. Maybe the latter is `IsIpfsEnabledForProfile`?
Or rename `IsIpfsDisabled` to `IsIpfsDisabledByFeatureOrPolicy`

Reply:
yep, thanks, done